packages/compass-indexes/src/modules/regular-indexes.ts

```diff
@@ -96,61 +86,103 @@ export const INITIAL_STATE: State = {
   error: null,
 };
 
-export default function reducer(state = INITIAL_STATE, action: AnyAction) {
-  if (isAction<IndexesAddedAction>(action, ActionTypes.IndexesAdded)) {
+export default function reducer(
+  state = INITIAL_STATE,
+  action: AnyAction
+): State {
+  if (
+    isAction<FetchIndexesStartedAction>(action, ActionTypes.FetchIndexesStarted)
+  ) {
     return {
       ...state,
-      indexes: action.indexes,
+      error: null,
+      isRefreshing: action.isRefreshing,
     };
   }
 
-  if (isAction<SetIsRefreshingAction>(action, ActionTypes.SetIsRefreshing)) {
+  if (
+    isAction<FetchIndexesSucceededAction>(
+      action,
+      ActionTypes.FetchIndexesSucceeded
+    )
+  ) {
+    // Merge the newly fetched indexes and the existing in-progress ones.
+    const inProgressIndexes = state.inProgressIndexes;
+    const allIndexes = _mergeInProgressIndexes(
+      action.indexes,
+      cloneDeep(inProgressIndexes)
+    );
+
     return {
       ...state,
-      isRefreshing: action.isRefreshing,
+      indexes: allIndexes,
+      isRefreshing: false,
     };
   }
 
-  if (isAction<SetErrorAction>(action, ActionTypes.SetError)) {
+  if (
+    isAction<FetchIndexesFailedAction>(action, ActionTypes.FetchIndexesFailed)
+  ) {
     return {
       ...state,
       error: action.error,
+      indexes: [],
```

Comment:
Hmmm, not sure if this is intended or not: if this is a failing refresh, should it remove all existing indexes from state or should it only set the error so all the previous indexes are still visible?

Reply:
It is intended. Not sure - does it matter? Fetching indexes shouldn't fail in regular use and the result is probably invalid if it ever does. You can always just hit refresh again?

Reply:
Fair enough! Just not a pattern I think we do in some other parts (like "old" documents in CRUD I think will stay on page if refresh failed), so wanted to double-check

Reply:
ok I'm going to change this back, then.  I don't actually have a strong preference now that I think about it and I don't want to do anything unexpected.